addons/ui/functions/fnc_setAdvancedElement.sqf

```diff
@@ -18,21 +18,28 @@
  * Public: No
  */
 
-params ["_element", "_show", ["_showHint", false, [true]], ["_force", false, [true]] ];
+params ["_element", "_show", ["_showHint", false, [true]], ["_force", false, [true]]];
 
 private _cachedElement = GVAR(configCache) getVariable _element;
-if (isNil "_cachedElement") exitWith {};
+if (isNil "_cachedElement") exitWith {TRACE_1("nil element",_this)};
 
 if (!_force && {!GVAR(allowSelectiveUI)}) exitWith {
+    TRACE_1("not allowed",_this);
     [LSTRING(Disallowed), 2] call EFUNC(common,displayTextStructured);
     false
 };
 
 _cachedElement params ["_idd", "_elements", "_location", "_conditions"];
 
 // Exit if main vehicle type condition not fitting
-private _canUseWeapon = ACE_player call CBA_fnc_canUseWeapon;
-if ((_canUseWeapon && {_location == 2}) || {!_canUseWeapon && {_location == 1}}) exitWith {false};
+private _canUseWeaponOrInCargo = ACE_player call CBA_fnc_canUseWeapon || {-1 < vehicle ACE_player getCargoIndex ACE_player};
+if (
+    (_canUseWeaponOrInCargo && {_location == VEHICLE_ONLY})
+    || {!_canUseWeaponOrInCargo && {_location == GROUND_ONLY}}
+) exitWith {
+    TRACE_3("skip location",_this,_canUseWeaponOrInCargo,_location);
+    false
+};
```

Comment:
I'd prefer this whole thing like this:
```sqf
if ((_canUseWeaponOrInCargo && {_location == VEHICLE_ONLY}) ||
    {!_canUseWeaponOrInCargo && {_location == GROUND_ONLY}}
) exitWith {
    TRACE_3("skip location",_this,_canUseWeaponOrInCargo,_location);
    false
};
```
Probably just because I am more used to it, either is mergable.